app/controllers/events_controller.rb

```diff
@@ -0,0 +1,26 @@
+class EventsController < ApplicationController
+  before_action :require_login, only: %i[new create]
+  def index
+    @past_events = Event.past
+    @future_events = Event.future
+  end
+
+  def new
+    @event = Event.new
+    @user = User.find(session[:current_user_id])
+  end
+
+  def create
+    user = User.find(session[:current_user_id])
+    event = user.created_events.build(event_params)
+    redirect_to event_url(event) if event.save
+  end
+
+  def show
+    @event = Event.find(params[:id])
+  end
+
+  def event_params
+    params.require(:event).permit(:title, :event_date, :description)
+  end
```

Comment:
Since this method is accessed only in this file, it is usually a good practice to make sure it's private
```
private

def event_params
...

```

Reply:
Moved the event_params method in the private interface of the Event Controllers class.